src/Symfony/Bundle/WebProfilerBundle/Controller/RouterController.php

```diff
@@ -99,4 +103,26 @@ private function getTraces(RequestDataCollector $request, $method)
 
         return $matcher->getTracesForRequest($traceRequest);
     }
+
+    /**
+     * Replaces scalars wrapped as Data instances by their actual values and removes other values.
+     */
+    private function filterScalarData(array $array)
+    {
+        foreach ($array as $key => $data) {
+            if (!$data instanceof Data) {
+                continue;
+            } elseif (is_scalar($value = $data->getRawData()[0][0])) {
+                $array[$key] = $value;
+            } elseif (!$value instanceof Stub) {
+                continue;
+            } elseif (is_scalar($value->value)) {
+                $array[$key] = $value->value;
+            } else {
+                unset($array[$key]);
```

Comment:
I think it's ok, but anyway, I updated the code to populate a new $result var.
Other comments addressed also.